### PR TITLE
feat: Preserve Splunk infra

### PR DESCRIPTION
**Pull request by rajkarkar-crest**
- Updated reusable workflow to v3.5
- Added opened, synchronize, reopened, and labeled types to run a workflow on all four events of a pull request.
- Added gh command to create **_preserve_infra_** label in TA repo.

### CODE REVIEW

File: enforce/.github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -9,14 +9,20 @@ on:
   pull_request:
     branches:
       - "**"
+    types:
+      - opened      # By default, a workflow runs when a pull_request event's activity is opened, synchronize, or reopened, if the types are not specified.
+      - reopened    # If types are specificed, the workflow will trigger only on those event types.
+      - synchronize # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+      - labeled     # Added opened, synchronize, reopened, and labeled to run workflow on all four event.
```
Comment (uoboda-splunk) [inline enforce/.github/workflows/build-test-release.yml:16]: I think comments might be removed

File: enforce/.github/workflows/build-test-release.yml [hunk 2]
```diff
@@ -9,14 +9,20 @@ on:
...
 {% if schedule_run is defined %}
   schedule:
     - cron: '0 2 * * *'
 {% endif %}
 
 jobs:
   call-workflow:
-    uses: splunk/addonfactory-workflow-addon-release/.github/workflows/reusable-build-test-release.yml@v3.4
+    if: github.event.action != 'labeled' || github.event.label.name == 'preserve_infra'
```
Comment (uoboda-splunk) [inline enforce/.github/workflows/build-test-release.yml:24]: To make a second run with preserved infra user would have to unlabeled and label again right?

Comment (rajkarkar-crest) [reply enforce/.github/workflows/build-test-release.yml:24]: Yes, correct. User and also retrigger the preserve infra workflow using Re-run jobs feature

File: enforce/.github/workflows/build-test-release.yml (hunk 1 above)
Comment (rajkarkar-crest) [reply enforce/.github/workflows/build-test-release.yml]: Removed comments.

**Review by uoboda-splunk — APPROVED** [review]

**Review by bhargav-nariyani-crest — APPROVED** [review]

**Pull request merged**